inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -376,6 +376,52 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
     }
 }
 
+void FrontEnd::processTrivialCases(const Model& model) {
+    std::map<ie::DataPtr, DataVector> vpuDataObjectsSortedByOrigData;
+    for (const auto& data : model->datas()) {
+        vpuDataObjectsSortedByOrigData[data->origData()].push_back(data);
+    }
+
+    std::vector<DataVector> trivialCases;
+    for (const auto& dataObjectsWithTheSameOrigData : vpuDataObjectsSortedByOrigData) {
+        if (dataObjectsWithTheSameOrigData.second.size() > 1) {
+            trivialCases.push_back(dataObjectsWithTheSameOrigData.second);
+        }
+    }
+
+    for (const auto& trivialCase : trivialCases) {
```

Comment:
Discussed offline. Try to re-implement approach by inserting copy stage before CNN network parsing and Convert stages insertion

Reply:
Done. It looks way better now.

Reply:
Do I understand correctly that you still insert copy stage after Convert stages insertion and parsing? If so, why?

Reply:
No, I've moved `processTrivialCases` call before `addDataTypeConvertStages` pass as we discussed.